### PR TITLE
Add support for RHEL Centos 8 RPM 

### PR DESCRIPTION
**Pull request by egieseke**
## Summary

Add packaging support for RHEL Centos 8 PRM.

## Test Plan

Testing with release pipeline.

### CODE REVIEW

**Comment by tsachiherman** [comment]
Just wondering - why wouldn't the Centos 7 installation work on Centos 8 ?

**Comment by onetechnical** [comment]
> Just wondering - why wouldn't the Centos 7 installation work on Centos 8 ?

Centos 8 has different dependencies (dnf-automatic instead of yum-cron).

**Comment by onetechnical** [comment]
> > Just wondering - why wouldn't the Centos 7 installation work on Centos 8 ?
> 
> Centos 8 has different dependencies (dnf-automatic instead of yum-cron).

Oh wait, I think I see what you're saying. Can't we just have one RPM that works with both distributions?

**Comment by onetechnical** [comment]
In answer to my own question, I think the answer is 'no', you can't have a single RPM that supports both distributions based on brief research.

File: docker/build/cicd.centos8.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+ARG ARCH="amd64"
+
+FROM ${ARCH}/centos:8
+ARG GOLANG_VERSION
+ARG ARCH="amd64"
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && \
```
Comment (onetechnical) [inline docker/build/cicd.centos8.Dockerfile:6]: Why do we `yum install` this but `dnf install` later? Shouldn't it all be dnf?

File: scripts/release/common/docker/centos8.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM centos:8
+
+WORKDIR /root
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
```
Comment (onetechnical) [inline scripts/release/common/docker/centos8.Dockerfile:4]: Same here, should we dnf everywhere?

File: docker/build/cicd.centos8.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
+ARG ARCH="amd64"
+
+FROM ${ARCH}/centos:8
```
Comment (onetechnical) [inline docker/build/cicd.centos8.Dockerfile:3]: I think this is fine, but I'm curious if there's a more supported stream image, e.g.:

quay.io/centos/centos:stream8

I don't actually know if that would be better or worse. ¯\_(ツ)_/¯

File: test/muleCI/Jenkinsfile [hunk 4]
```diff
@@ -1,3 +1,3 @@
-@Library('go-algorand-ci') _
+@Library('go-algorand-ci@update-cicd-cachename') _
```
Comment (onetechnical) [inline test/muleCI/Jenkinsfile:1]: Don't forget to update this.

**Review by onetechnical — COMMENTED** [review]
Most of this looks alright, had some comments.

Also, I'm curious how this will work with repositories. I don't know of a good way to test this.

File: docker/build/cicd.centos8.Dockerfile (hunk 3 above)
Comment (bricerisingalgorand) [reply docker/build/cicd.centos8.Dockerfile:3]: oh right, i thought we were going to use that

File: docker/build/cicd.centos8.Dockerfile (hunk 1 above)
Comment (bricerisingalgorand) [reply docker/build/cicd.centos8.Dockerfile:6]: it shouldn't actually make a difference, but it wouldn't hurt to switch to dnf

File: package.yaml [hunk 5]
```diff
@@ -21,6 +21,17 @@ agents:
       - VERSION=$VERSION
     workDir: $HOME/projects/go-algorand
 
+  - name: rpm.centos8
+    dockerFilePath: docker/build/cicd.centos8.Dockerfile
+    image: algorand/go-algorand-ci-linux-centos8
+    version: scripts/configure_dev-deps.sh
```
Comment (bricerisingalgorand) [inline package.yaml:27]: does this file still exist? I thought tsachi was removing it

Comment (tsachiherman) [reply package.yaml:27]: I cleared it's content, but kept it around so it won't create build failures.

**Comment by bricerisingalgorand** [comment]
Assuming it works I think this looks ok to me. It doesn't appear to include publishing to the rpm repo, but I think that's ok for this story

**Review by bricerisingalgorand — DISMISSED** [review]

**Comment by bricerisingalgorand** [comment]
I gave it a tentative approval as long as John's feedback is resolved

File: docker/build/cicd.centos8.Dockerfile (hunk 1 above)
Comment (egieseke) [reply docker/build/cicd.centos8.Dockerfile:6]: Replaced use of yum with dnf.

**Comment by onetechnical** [comment]
> Assuming it works I think this looks ok to me. It doesn't appear to include publishing to the rpm repo, but I think that's ok for this story

Oh, you know what, I believe we lock in on `s3://$PREFIX/algorand-$VERSION-1.x86_64.rpm` and `s3://$PREFIX/algorand-devtools-$VERSION-1.x86_64.rpm` in `scripts/release/mule/deploy/rpm/deploy.sh`, so we should be fine. I think we'll need to add a story later to add this to repo properly, once we thoroughly test.

File: scripts/release/common/docker/centos8.Dockerfile (hunk 2 above)
Comment (egieseke) [reply scripts/release/common/docker/centos8.Dockerfile:4]: Updated to use dnf.

File: test/muleCI/Jenkinsfile (hunk 4 above)
Comment (egieseke) [reply test/muleCI/Jenkinsfile:1]: Removed branch.

File: docker/build/cicd.centos8.Dockerfile (hunk 3 above)
Comment (egieseke) [reply docker/build/cicd.centos8.Dockerfile:3]: Updated to pull from quay.io/centos/centos:stream8.

File: docker/build/cicd.centos8.Dockerfile (hunk 1 above)
Comment (egieseke) [reply docker/build/cicd.centos8.Dockerfile:6]: updated to use dnf.

File: scripts/release/common/setup.sh [hunk 6]
```diff
@@ -105,6 +105,7 @@ fi
 
 sudo usermod -a -G docker ubuntu
 sg docker "docker pull centos:7"
+sg docker "docker pull centos:8"
```
Comment (onetechnical) [inline scripts/release/common/setup.sh:108]: Might want to replace this with the stream image, but not really a blocker.

**Review by onetechnical — APPROVED** [review]
Provided this passes testing, I'm okay with this.

**Comment by onetechnical** [comment]
This PR is still in draft mode, though.

File: scripts/release/common/setup.sh (hunk 6 above)
Comment (egieseke) [reply scripts/release/common/setup.sh]: Updated to pull quay.io/centos/centos:stream8.

**Review by onetechnical — APPROVED** [review]
This still looks fine to me, though I'd like to see it exercised to verify it works.

**Comment by algobarb** [comment]
Approved. Good work incorporating all the other suggestions!

**Review by algobarb — APPROVED** [review]

**Pull request merged**